.ci/bindist/linux/snap/Dockerfile

```diff
@@ -0,0 +1,67 @@
+# This Dockerfile is:
+#
+#   https://github.com/snapcore/snapcraft/pull/3384
+#   + xenial => focal
+#
+FROM ubuntu:focal as builder
+
+# Grab dependencies
+RUN apt-get update
+RUN apt-get dist-upgrade --yes
+RUN apt-get install --yes \
+      curl \
+      jq \
+      squashfs-tools
+
+# Grab the core snap (for backwards compatibility) from the stable channel and
+# unpack it in the proper place.
+RUN curl -L $(curl -H 'X-Ubuntu-Series: 16' 'https://api.snapcraft.io/api/v1/snaps/details/core' | jq '.download_url' -r) --output core.snap
+RUN mkdir -p /snap/core
+RUN unsquashfs -d /snap/core/current core.snap
```

Comment:
I think each of these three should live in a single layer, they seem like one operation conceptually